src/librustc_typeck/diagnostics.rs

```diff
@@ -1611,9 +1611,9 @@ static BAR: _ = "test"; // error, explicitly write out the type instead
 "##,
 
 E0122: r##"
-An attempt was made to add a generic constraint to a type alias. While Rust will
-allow this with a warning, it will not currently enforce the constraint.
-Consider the example below:
+An attempt was made to add a generic constraint to a type alias. This constraint
+is entirely ignored. For backwards compatibility, Rust still allows this with a
+warning. Consider the example below:
```

Comment:
> the constraint is only ignored when referencing the alias, not within it.

Are you sure?

This is accepted:
```rust
struct Test<T: Copy> { t : T}

type MyTest<T> = Test<T>;
```

And so it this:
```rust
trait Test {
    type Type;
}

type MyTest<T> = <T as Test>::Type;
```

To me it looks like constraints are not at all checked or enforced inside aliases.